guards/functional.py

```diff
@@ -0,0 +1,75 @@
+from functools import wraps
```

Comment:
I'm not sure why but my `pylint` screams at this row and all the imports are red because of this, so I had to turn it down in `.vscode/settings.json`  
You can provide a good linter setting for this project (I'm not sure why is that egg-info for but maybe some of the development settings should not be gitignored)

Reply:
It's strange. My linter does not scream at all. It may be a global vscode configuration issue or stg like that. I think it is better to keep the repo independent of repositories. 

The egg-info is to make it not pollute the repository when installing with setup.py